File: packages/posthog/android/src/main/java/io/capawesome/capacitorjs/plugins/posthog/classes/options/SetupOptions.java
package io.capawesome.capacitorjs.plugins.posthog.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class SetupOptions {

    @NonNull
    private String apiKey;

    @NonNull
    private String apiHost;

    @Nullable
    private Boolean enableSessionReplay;

    @Nullable
    private Boolean optOut;

    @Nullable
    private SessionReplayOptions sessionReplayConfig;

    public SetupOptions(@NonNull String apiKey, @NonNull String apiHost) {
        this.apiKey = apiKey;
        this.apiHost = apiHost;
    }

    @NonNull
    public String getApiKey() {
        return apiKey;
    }

    @NonNull
    public String getApiHost() {
        return apiHost;
    }

    public boolean getEnableSessionReplay() {
        return enableSessionReplay != null ? enableSessionReplay : false;
    }

    public void setEnableSessionReplay(boolean enableSessionReplay) {
        this.enableSessionReplay = enableSessionReplay;
    }

    public boolean getOptOut() {
        return optOut != null ? optOut : false;
    }

    public void setOptOut(boolean optOut) {
        this.optOut = optOut;
    }

    @Nullable
    public SessionReplayOptions getSessionReplayConfig() {
        return sessionReplayConfig;
    }

    public void setSessionReplayConfig(@Nullable SessionReplayOptions sessionReplayConfig) {
        this.sessionReplayConfig = sessionReplayConfig;
    }
}

File: packages/posthog/ios/Plugin/Classes/Options/SetupOptions.swift
import Foundation

@objc public class SetupOptions: NSObject {
    private var apiKey: String
    private var apiHost: String
    private var enableSessionReplay: Bool
    private var optOut: Bool
    private var sessionReplayConfig: SessionReplayOptions?

    init(apiKey: String, apiHost: String, enableSessionReplay: Bool, optOut: Bool, sessionReplayConfig: [String: Any]?) {
        self.apiKey = apiKey
        self.apiHost = apiHost
        self.enableSessionReplay = enableSessionReplay
        self.optOut = optOut

        if let config = sessionReplayConfig {
            self.sessionReplayConfig = SessionReplayOptions(
                screenshotMode: config["screenshotMode"] as? Bool ?? false,
                maskAllTextInputs: config["maskAllTextInputs"] as? Bool ?? true,
                maskAllImages: config["maskAllImages"] as? Bool ?? true,
                maskAllSandboxedViews: config["maskAllSandboxedViews"] as? Bool ?? true,
                captureNetworkTelemetry: config["captureNetworkTelemetry"] as? Bool ?? true,
                debouncerDelay: config["debouncerDelay"] as? Double ?? 1.0
            )
        }
    }

    func getApiKey() -> String {
        return apiKey
    }

    func getApiHost() -> String {
        return apiHost
    }

    func getEnableSessionReplay() -> Bool {
        return enableSessionReplay
    }

    func getOptOut() -> Bool {
        return optOut
    }

    func getSessionReplayConfig() -> SessionReplayOptions? {
        return sessionReplayConfig
    }
}
